File: LeetCode/JavaScript/1051-Height-Checker.js
// 1051. Height Checker
// https://leetcode.com/problems/height-checker/

/**
 * @param {number[]} heights
 * @return {number}
 */

var heightChecker = function (heights) {
  let count = 0;
  const sortedHeights = heights.slice().sort((a, b) => a - b);

  for (let i = 0; i < heights.length; i++) {
    if (heights[i] !== sortedHeights[i]) count++;
  }

  return count++;
};

File: LeetCode/Python/1051_Height_Checker.py
# 1051. Height Checker
# https://leetcode.com/problems/height-checker/


class Solution:
    def heightChecker(self, heights: list[int]) -> int:
        count = [0] * 101
        for h in heights:
            count[h] += 1

        mismatch_count = 0
        current_height = 1

        for h in heights:
            while count[current_height] == 0:
                current_height += 1

            if h != current_height:
                mismatch_count += 1

            count[current_height] -= 1

        return mismatch_count
